cloudinit/net/ephemeral.py

```diff
@@ -76,9 +74,20 @@ def __init__(
         self.cleanup_cmds: List[Callable] = []
         self.distro = distro
         self.cidr = f"{self.ip}/{self.prefix}"
+        self.interface_addrs_before_dhcp = interface_addrs_before_dhcp.get(
+            self.interface, {}
+        )
+        self.interface_addrs_after_dhcp = netinfo.netdev_info().get(
```

Comment:
Nit: Is there a reason to make this an instance variable rather than putting it in `_bringup_device()`?